src/test/java/org/folio/rest/api/CirculationSettingsAPITest.java

```diff
@@ -1,27 +1,53 @@
 package org.folio.rest.api;
 
-import static org.hamcrest.CoreMatchers.is;
-import static org.hamcrest.junit.MatcherAssert.assertThat;
+import io.vertx.core.json.JsonObject;
+import org.folio.rest.support.ApiTests;
+import org.folio.rest.support.http.AssertingRecordClient;
+import org.folio.rest.support.http.InterfaceUrls;
+import org.junit.Before;
+import org.junit.Test;
 
 import java.net.MalformedURLException;
 import java.util.UUID;
 import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeoutException;
 
-import org.folio.rest.support.ApiTests;
-import org.folio.rest.support.http.AssertingRecordClient;
-import org.folio.rest.support.http.InterfaceUrls;
-import org.junit.Test;
-
-import io.vertx.core.json.JsonObject;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.junit.MatcherAssert.assertThat;
 
 public class CirculationSettingsAPITest extends ApiTests {
+  private static final String TABLE_NAME = "circulation_settings";
 
   private final AssertingRecordClient circulationSettingsClient =
     new AssertingRecordClient(
-    client, StorageTestSuite.TENANT_ID, InterfaceUrls::circulationSettingsUrl,
+      client, StorageTestSuite.TENANT_ID, InterfaceUrls::circulationSettingsUrl,
       "circulation-settings");
 
+  @Before
+  public void beforeEach() {
+    StorageTestSuite.cleanUpTable(TABLE_NAME);
+  }
+
+  @Test
+  public void updateInsteadCreateWithTheSameName() throws MalformedURLException,
+    ExecutionException, InterruptedException, TimeoutException {
+
+    String id = UUID.randomUUID().toString();
+    JsonObject circulationSettingsJson = getCirculationSetting(id);
+    JsonObject circulationSettingsResponse =
+      circulationSettingsClient.create(circulationSettingsJson).getJson();
+    String updatedId = UUID.randomUUID().toString();
+    JsonObject circulationSettingsJsonUpdated = getCirculationSetting(updatedId);
+    JsonObject updatedValue = new JsonObject().put("sample", "OK1");
```

Comment:
Let's refactor these tests a little, I forgot to do this when I was working on circulation settings. Let's move most of the string literals to constants and give them different values. Like "setting-name", "setting-key", "setting-value", "updated-setting-value" instead of "sample", "OK", "OK1" etc. Method getCirculationSetting() needs to be updated too, as well as a couple of other tests.

Reply:
refactored